connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1906,6 +1917,8 @@ private void reconfigureConnector(final String connName, final Callback<Void> cb
                 if (isLeader()) {
                     writeToConfigTopicAsLeader(() -> configBackingStore.putTaskConfigs(connName, rawTaskProps));
                     cb.onCompletion(null, null);
+                } else if (restClient == null) {
+                    throw new NotLeaderException("Request forwarding disabled in distributed MirrorMaker2; reconfiguring tasks must be performed by the leader", leaderUrl());
```

Comment:
Similar wording suggestions for this message:
```suggestion
                    // TODO: Update this message if KIP-710 is accepted and merged
                    //       (https://cwiki.apache.org/confluence/display/KAFKA/KIP-710%3A+Full+support+for+distributed+mode+in+dedicated+MirrorMaker+2.0+clusters)
                    throw new NotLeaderException("This worker is not able to communicate with the leader of the cluster, "
                                    + "which is required for dynamically-reconfiguring connectors. If running MirrorMaker 2 "
                                    + "in dedicated mode, consider deploying the connectors for MirrorMaker 2 directly onto a "
                                    + "distributed Kafka Connect cluster.",
                            leaderUrl()
                    );
```